### PR TITLE
Split RecordType to type and status

### PR DESCRIPTION
**Pull request by JiayuZzz**
### What problem does this PR solve?

Split RecordType to type and status for flexibility

### What is changed and how it works?

Split RecordType to DataType (No delete or dirty record type) and status (Normal, dirty, outdated) for further modification on cleaner

### Check List 

Tests 

- Unit test
- Integration test

Side effects

no

### CODE REVIEW

File: engine/generic_list.hpp [hunk 1]
```diff
@@ -182,7 +182,8 @@ class GenericList final : public Collection {
       }
       kvdk_assert(Collection::ExtractID(curr->Key()) == owner->ID(), "");
       kvdk_assert(
-          (curr->entry.meta.type == DataType && curr->Validate()) || dirty(),
+          (curr->entry.meta.mark.record_type == ElemType && curr->Validate()) ||
```
Comment (ZiyanShi) [inline engine/generic_list.hpp:185]: Too many layers.
`record.type` is much simpler than `record.entry.meta.mark.record_type`

File: engine/generic_list.hpp [hunk 2]
```diff
@@ -514,7 +515,7 @@ class GenericList final : public Collection {
     PMemOffsetType prev_off = (prev == Head()) ? NullPMemOffset : prev.Offset();
     PMemOffsetType next_off = (next == Tail()) ? NullPMemOffset : next.Offset();
     DLRecord* record = DLRecord::PersistDLRecord(
-        addressOf(space.offset), space.size, timestamp, DataType,
+        addressOf(space.offset), space.size, timestamp, ElemType,
```
Comment (ZiyanShi) [inline engine/generic_list.hpp:518]: `ElemType` is not `RecordMark`

File: engine/generic_list.hpp (hunk 1 above)
Comment (JiayuZzz) [reply engine/generic_list.hpp:185]: This should be record.GetMark().type. We need to store record type in DataEntry. 
Maybe we don't need RecordMark, just two fields RecordType and RecordStatus

Comment (KVKIT) [reply engine/generic_list.hpp]: Support to use RecordType and RecordStatus instead of RecordMark as its meaning is not much obvious.

**Review by ZiyanShi — APPROVED** [review]

**Pull request merged**